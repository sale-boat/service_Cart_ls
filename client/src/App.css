.container{
	display : flex;
	width : 244px;
	height : 500px; 
	flex-direction : column;
	background-color : white;

	border: 1px #D3D3D3 solid;
	padding: 10px 20px; 
    border-radius: 4px;
}

.priceChris{
	padding-top:5px;
	color : #B12704;
	font-size: 17px;
	font-family: Lato;
}

.freeshipping{
	margin-top : 10px;
	font-size : 13px;
	line-height : 1.6;
	font-family: Lato;
}


.tooltip{
	position : relative;
	display : inline-block;
	color : #0066c0;
    cursor : pointer;
}

.tooltip .tooltiptext{
	visibility: hidden;
	width: 200px;
	background-color: white;
	color: black;
	z-index: 1;
	position: absolute;
	opacity: 0;
	transition:opacity 0.3s;
}

.tooltip:hover .tooltiptext{
	visibility: visible;
	opacity: 1;
}


/* .body{
    position: absolute;
    top:0;
	left:0;
	bottom:0;
	right:0;
	height:100%;
	width:100%; 
	border: 1px solid white;
} */