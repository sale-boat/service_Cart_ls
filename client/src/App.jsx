import React from 'react';
import CSSModules from 'react-css-modules';
import app from './App.css';
import axios from 'axios';
import DetailsBox from './DetailsBox.jsx';
import Dates from './Dates.jsx';
import Buttons from './Buttons.jsx';
import DeliverModal from './DeliverModal.jsx';
import AddToListModal from './AddToListModal.jsx';


class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      // price: '',
    };
    this.child = React.createRef();
    this.clickOutside = this.clickOutside.bind(this);
    this.getData = this.getData.bind(this);
  }


  componentDidMount() {
    // const urlArray = document.URL.split('/');
    // const productID = Number(urlArray[urlArray.length - 1]);
    // axios.get(`/api/product/${productID}`)
    //   .then((res) => {
    //     this.setState({ price: res.data[0].price });
    //   })
    //   .catch((err) => {
    //     console.log(err);
    //   });

    this.getData();
  }

  getData() {
    const urlArray = document.URL.split('/');
    const productID = Number(urlArray[urlArray.length - 1]);
    axios.get(`/api/product/${productID}`)
      .then((res) => {
        this.setState({ price: res.data[0].price });
      })
      .catch((err) => {
        console.log(err);
      });
  }

  clickOutside(event) {
    if (event.target.className === 'body' || event.target.className === 'container') {
      this.child.current.setState({ clicked: false });
    }
  }

  render() {
    const { price } = this.state;
    const style = { display: price ? 'flex' : 'none' };
    return (
      <div className={app.container} style={style}>
        <div className={app.priceChris}>
          {' '}
          $
          { price }
          {' '}
        </div>
        <div className={app.freeshipping}>
            &
          {' '}
          <b>FREE Shipping</b>
          {' '}
          on orders over $25 shipped by Amazon.
          {' '}
          <DetailsBox ref={this.child} />
          <Dates />
          <Buttons />
          <DeliverModal />
          <AddToListModal />
        </div>
      </div>

    );
  }
}

export default App;
