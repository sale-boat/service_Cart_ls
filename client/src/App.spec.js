import React from 'react';
import { shallow } from 'enzyme';
import App from './App.jsx';

describe('App Component', () => {
  it('should render', () => {
    const app = shallow(<App />);
    expect(app).toMatchSnapshot();
  });
});
