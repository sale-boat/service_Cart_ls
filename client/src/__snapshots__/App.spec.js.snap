// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render 1`] = `
<div
  className="body"
  onClick={[Function]}
>
  <div
    className="container"
  >
    <div
      className="price"
    >
       $
       
    </div>
    <div
      className="freeshipping"
    >
      &
       
      <b>
        FREE Shipping
      </b>
       
      on orders over $25 shipped by Amazon.
       
      <DetailsBox />
      <Dates />
      <Buttons />
      <DeliverModal />
      <AddToListModal />
    </div>
  </div>
</div>
`;
